src/edu/csus/ecs/pc2/ui/EditProblemPane.java

```diff
@@ -1494,15 +1517,32 @@ private VALIDATOR_TYPE getValidatorTypeFromUI() {
         return validatorType;
     }
 
+    private CUSTOM_VALIDATOR_TYPE getCustomValidatorTypeFromUI() {
```

Comment:
As noted in the comment about `enum CUSTOM_VALIDATOR_TYPE` above, this method isn't really returning a "custom validator TYPE"; it's returning an indication of the type of _interface_ used by the custom validator.    (I imagine for example that in the future we might have multiple different types of custom validators, of which "interactive" is just one type -- perhaps we would have CUSTOM_VALIDATOR_TYPE.HIGH_EFFICIENCY and CUSTOM_VALIDATOR_TYPE.MANUAL, for example -- the latter being a validator which operates by communicating with a human through the Judge's keyboard.  In that case, we wouldn't want the enum named `CUSTOM_VALIDATOR_TYPE`  to be returning the _type of interface used_; we'd want it to return the _type of the validator itself_.)

I think the method should be named (something like) `getCustomValidatorInterfaceTypeFromUI()` (or maybe a shorter form -- but something that makes it clear it's returning the type of the _interface used by the validator_, not the actual _type of the validator_.